vector/src/main/java/im/vector/VectorApp.java

```diff
@@ -352,6 +359,27 @@ public void onActivityDestroyed(Activity activity) {
         registerReceiver(mLanguageReceiver, new IntentFilter(Intent.ACTION_LOCALE_CHANGED));
         registerReceiver(mLanguageReceiver, new IntentFilter(Intent.ACTION_CONFIGURATION_CHANGED));
 
+        // Register the registration receiver to the local broadcast manager.
```

Comment:
After reading some documentation about it, the LocalBroadcastManager.registerReceiver() should be call in Activities that actually send broadcasts.
Please move this code in a protected method in VectorAppCompatActivity and call it in the onResume() method of the related Activities. mRegistrationReceiver and mUniversalLinkReceiver can be lazy loaded private members of VectorAppCompatActivity.
In the onPause() method of the Activity, you will be able to unregister the receiver then.
Note that what is done in initUiAndData() of such Activity should be moved to the onResume() method then.